ChatBot/ChatBot/Controller/ChatRoomViewController.swift

```diff
@@ -0,0 +1,191 @@
+        textView.layer.borderColor = UIColor.systemGray.cgColor
+        textView.layer.cornerRadius = 12
+        return textView
+    }()
+    
+    private lazy var sendButton: UIButton = {
+        let button = UIButton()
+        button.layer.cornerRadius = 8
+        button.backgroundColor = #colorLiteral(red: 0.7890922427, green: 0.9981873631, blue: 0.9562725425, alpha: 1)
+        button.setImage(UIImage(systemName: "paperplane"), for: .normal)
+        button.addTarget(self, action: #selector(tappedSendButton), for: .touchUpInside)
+        return button
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configurationUI()
+        configurationCell()
+        setUpSnapshot()
+        self.textInputView.delegate = self
+        NotificationCenter.default.addObserver(self, selector: #selector(updateCollectionView), name: Notification.Name("ChatsUpdatedNotification"), object: nil)
+    }
+    
+    private func configurationUI() {
+        view.backgroundColor = .white
+        view.addSubview(mainStackView)
+        
+        mainStackView.addArrangedSubview(collectionView)
+        mainStackView.addArrangedSubview(userInputStackView)
+        
+        userInputStackView.addArrangedSubview(textInputView)
+        userInputStackView.addArrangedSubview(sendButton)
+        
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        mainStackView.translatesAutoresizingMaskIntoConstraints = false
+        userInputStackView.translatesAutoresizingMaskIntoConstraints = false
+        textInputView.translatesAutoresizingMaskIntoConstraints = false
+        sendButton.translatesAutoresizingMaskIntoConstraints = false
+        
+        let safeLayoutGuide = view.safeAreaLayoutGuide
+        
+        NSLayoutConstraint.activate([
+            mainStackView.topAnchor.constraint(equalTo: safeLayoutGuide.topAnchor),
+            mainStackView.bottomAnchor.constraint(equalTo: view.keyboardLayoutGuide.topAnchor),
+            mainStackView.leadingAnchor.constraint(equalTo: safeLayoutGuide.leadingAnchor, constant: 12),
+            mainStackView.trailingAnchor.constraint(equalTo: safeLayoutGuide.trailingAnchor, constant: -12),
+            textInputView.widthAnchor.constraint(equalTo: userInputStackView.widthAnchor, multiplier: 0.8),
+            textInputView.heightAnchor.constraint(equalToConstant: textInputView.estimatedHeight),
+            sendButton.widthAnchor.constraint(equalTo: userInputStackView.widthAnchor, multiplier: 0.15),
+        ])
+    }
+    
+    @objc func tappedSendButton() {
+        textInputView.resignFirstResponder()
+        
+        guard let question = textInputView.text, question != "" else { return }
+        //        chatManager.appendChat(question: question)
+        chatManager.appendQuestion(question: question) {
+            sendButton.isEnabled = false
+            
+        }
+        
+        chatManager.sendQuestion {
```

Comment:
```suggestion
        chatManager.appendQuestion(question: question)
        
        sendButton.isEnabled = false
        chatManager.sendQuestion {
```

=> 구두로 말씀드린대로 sendQuestion 전에 sendButton 비활성화 처리 하면 되지 않나 생각합니다. 
appendQuestion 에는 completionHandler 가 굳이 필요없어 보이는데 한번 확인 부탁드릴게요!

Reply:
말씀하신 대로 수정하였습니다.